java-frontend/src/main/java/org/sonar/java/se/checks/MapComputeIfAbsentOrPresentCheck.java

```diff
@@ -126,15 +151,21 @@ private static boolean isMethodInvocationThrowingCheckedException(ExpressionTree
   }
 
   private static boolean isInsideIfStatementWithNullCheckWithoutElse(MethodInvocationTree mit) {
+    return getIfStatementParent(mit).map(ifStatementTree -> ifStatementTree.elseStatement() == null
```

Comment:
From how it is implemented, `getIfStatementParent(mit)` can be pretty costly (it potentially explore the whole AST), and we are going to make such calls for each `map.containsKey()` or `map.get()`.

What about precomputing them or coaching them to avoid having to reexplore the AST for all these invocations?